opslevel/datasource_opslevel_user.go

```diff
@@ -0,0 +1,98 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure UserDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &UserDataSource{}
+
+func NewUserDataSource() datasource.DataSource {
+	return &UserDataSource{}
+}
+
+// UserDataSource manages a User data source.
+type UserDataSource struct {
+	CommonDataSourceClient
+}
+
+// UserDataSourceModel describes the data source data model.
+type UserDataSourceModel struct {
+	Email      types.String `tfsdk:"email"`
+	Id         types.String `tfsdk:"id"`
+	Identifier types.String `tfsdk:"identifier"`
+	Name       types.String `tfsdk:"name"`
+	Role       types.String `tfsdk:"role"`
+}
+
+func NewUserDataSourceModel(ctx context.Context, user opslevel.User, identifier string) UserDataSourceModel {
+	return UserDataSourceModel{
+		Email:      types.StringValue(user.Email),
+		Id:         types.StringValue(string(user.Id)),
+		Identifier: types.StringValue(identifier),
+		Name:       types.StringValue(user.Name),
+		Role:       types.StringValue(string(user.Role)),
```

Comment:
I am really surprised that opslevel-go doesn't support teams. (Not a suggestion/complaint - I expected the datasource to also give me a list of teams the user is involved in since that's probably the most useful piece of info. However it's not possible right now.)